docs/content/concepts/assets/asset-checks/checking-for-data-freshness.mdx

```diff
@@ -0,0 +1,339 @@
+
+---
+
+## How it works
+
+An asset is considered fresh if it's been materialized or observed within a defined time window. Otherwise, assets are considered overdue and in need of an update. To identify these assets, you can use **freshness checks**.
+
+Let's say an asset begins materializing at 12:00AM and takes 10 minutes to complete. Allowing for operational constraints and delays, the asset should always be materialized by 12:30AM. Using a freshness check, we can define the time that the asset should be updated. If it's not, Dagster will mark it as **Overdue**.
+
+How freshness checks are implemented depends on the type of asset they target:
+
+- **For source assets**, which are assets whose data feeds a Dagster pipeline, freshness checks utilize the time the asset was last updated to determine freshness. In this guide, we'll demonstrate how to use [observable source assets](/concepts/assets/asset-observations#observable-source-assets) and a [schedule](/concepts/partitions-schedules-sensors/schedules) to achieve this.
+
+- **For materializable assets**, which are assets materialized by Dagster, Dagster can infer the asset's last update time using the asset's latest materialization timestamp. In this guide, we'll demonstrate how to use <PyObject object="asset" decorator />-decorated assets and a [sensor](/concepts/partitions-schedules-sensors/sensors) to achieve this.
+
+For materializable assets, freshness checks can also help you communicate SLAs for their data freshness. For example, downstream consumers of the assets can look at the checks that are defined on those assets to determine when and how often they’re expected to be updated.
+
+---
+
+## Prerequisites
+
+Before continuing, you should be familiar with:
+
+- [Software-defined Assets](/concepts/assets/software-defined-assets)
+- [Asset checks](/concepts/assets/asset-checks)
+- [Schedules](/concepts/partitions-schedules-sensors/schedules) and/or [sensors](/concepts/partitions-schedules-sensors/sensors)
+
+---
+
+## Defining freshness checks for source assets
+
+In this section, we'll demonstrate how to implement freshness checks for source assets. Source assets are assets whose data feeds a Dagster pipeline, but isn't materialized by Dagster.
+
+### Step 1: Track the asset's last update time
+
+To run freshness checks on source assets, the checks need to know when the source assets were last updated. [Observable source assets](/concepts/assets/asset-observations#observable-source-assets) can be used to track the update times of these assets.
+
+The following is an example of multiple Snowflake tables, backed by an observation function that queries Snowflake to find the most recent time the tables were updated. The function yields the update time as metadata to be stored in the Dagster event log:
+
+```python
+from dagster import multi_observable_source_asset, AssetSpec, Definitions, MetadataValue
```

Comment:
We should get these examples under test. After this gets merged, I or @dpeng817 should do a pass to extract them out into .py files.

Reply:
Agreed - I left a note about this to make sure it was accounted for. Additionally, I think I saw a note in the original Notion draft you wrote that a better example should be used for the materializable asset section - d'you think we can get that in and then do a fast-follow for moving everything?